### PR TITLE
Feature/separate fork config re config

### PR DESCRIPTION
**Pull request by talekhinezh**
* Refactor Fork configurations

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/service/SystemConfigService.java [hunk 1]
```diff
@@ -354,7 +354,7 @@ static JSONObject prepareRadixEngineConfiguration(
 		forkConfigTreeMap.forEach((e, config) -> forks.put(
 			jsonObject()
 				.put("name", config.getName())
-				.put("ceilingView", config.getEpochCeilingView().number())
+				.put("maxRounds", config.getConfig().getMaxRounds())
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/service/SystemConfigService.java:357]: I'll update open rpc spec with this change.

**Review by siy — APPROVED** [review]
LGTM

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/ForksModule.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		OptionalBinder.newOptionalBinder(binder(), new TypeLiteral<UnaryOperator<Set<ForkConfig>>>() { });
+		install(new BetanetForkTxnRulesModule());
+	}
+
+	@Provides
+	@Singleton
+	private Forks forks(
+		Set<ForkConfig> forkConfigs,
+		Optional<UnaryOperator<Set<ForkConfig>>> transformer,
+		Map<String, Function<RERulesConfig, RERules>> rules
```
Comment (iamyulong) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/ForksModule.java:47]: Minor: can we have a simple `Map<ForkConfig, RERules>`?

Comment (siy) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/ForksModule.java:47]: I think using function here is more flexible approach.

**Pull request merged**